crates/sui-core/src/consensus_handler.rs

```diff
@@ -646,238 +740,1639 @@ impl<C: CheckpointServiceNotify + Send + Sync> ConsensusHandler<C> {
+    }
+
+    fn load_deferred_transactions(
+        &self,
+        state: &mut CommitHandlerState,
+        commit_info: &ConsensusCommitInfo,
+    ) -> (
+        Vec<VerifiedExecutableTransaction>,
+        Vec<VerifiedExecutableTransaction>,
+        HashMap<TransactionDigest, DeferralKey>,
+    ) {
+        let mut previously_deferred_tx_digests = HashMap::new();
+
+        // DONE(commit-handler-rewrite): Load transactions deferred from previous commits, compute the digest set of all such transactions.
+        let deferred_txs: Vec<_> = self
+            .epoch_store
+            .load_deferred_transactions_for_up_to_consensus_round_v2(
+                &mut state.output,
+                commit_info.round,
+            )
+            .expect("db error")
+            .into_iter()
+            .flat_map(|(key, txns)| txns.into_iter().map(move |tx| (key, tx)))
+            .map(|(key, tx)| {
+                previously_deferred_tx_digests.insert(*tx.digest(), key);
+                tx
+            })
+            .collect();
+        trace!(
+            "loading deferred transactions: {:?}",
+            deferred_txs.iter().map(|tx| tx.digest())
+        );
+
+        // DONE(commit-handler-rewrite): load all deferred randomness-using txns
+        let deferred_randomness_txs = if state.dkg_failed || state.randomness_round.is_some() {
+            let txns: Vec<_> = self
+                .epoch_store
+                .load_deferred_transactions_for_randomness_v2(&mut state.output)
+                .expect("db error")
+                .into_iter()
+                .flat_map(|(key, txns)| txns.into_iter().map(move |tx| (key, tx)))
+                .map(|(key, tx)| {
+                    previously_deferred_tx_digests.insert(*tx.digest(), key);
+                    tx
+                })
+                .collect();
+            trace!(
+                "loading deferred randomness transactions: {:?}",
+                txns.iter().map(|tx| tx.digest())
+            );
+            txns
+        } else {
+            vec![]
+        };
+
+        (
+            deferred_txs,
+            deferred_randomness_txs,
+            previously_deferred_tx_digests,
+        )
+    }
+
+    fn init_congestion_tracker(
+        &self,
+        commit_info: &ConsensusCommitInfo,
+        for_randomness: bool,
+        txns: &[VerifiedExecutableTransaction],
+    ) -> SharedObjectCongestionTracker {
+        SharedObjectCongestionTracker::from_protocol_config(
+            self.epoch_store
+                .consensus_quarantine
+                .read()
+                .load_initial_object_debts_v2(
+                    &self.epoch_store,
+                    commit_info.round,
+                    for_randomness,
+                    txns,
+                )
+                .expect("db error"),
+            self.epoch_store.protocol_config(),
+            for_randomness,
+        )
+    }
+
+    fn init_randomness(
```

Comment:
It seems that this function should be a struct function of `CommitHandlerState`, that takes `epoch_store` as param. Similar for a few other functions. When that grows big enough we might want to move it to a separate file.